File: app/src/main/java/at/shockbytes/corey/core/receiver/NotificationReceiver.java
package at.shockbytes.corey.core.receiver;

import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;

import javax.inject.Inject;

import at.shockbytes.corey.core.CoreyApp;
import at.shockbytes.corey.common.core.util.ResourceManager;
import at.shockbytes.corey.util.schedule.ScheduleManager;

public class NotificationReceiver extends BroadcastReceiver {

    @Inject
    protected ScheduleManager manager;

    @Override
    public void onReceive(Context context, Intent intent) {
        ((CoreyApp) context.getApplicationContext()).getAppComponent().inject(this);

        if (manager.isWeighNotificationDeliveryEnabled() &&
                ResourceManager.getDayOfWeek() == manager.getDayOfWeighNotificationDelivery()) {
            manager.postWeighNotification();
        }
        if (manager.isWorkoutNotificationDeliveryEnabled()) {
            manager.tryPostWorkoutNotification();
        }
    }

}

File: app/src/main/java/at/shockbytes/corey/core/receiver/NotificationReceiver.kt
package at.shockbytes.corey.core.receiver

import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent
import at.shockbytes.corey.core.CoreyApp
import at.shockbytes.corey.data.reminder.ReminderManager
import timber.log.Timber
import javax.inject.Inject

class NotificationReceiver : BroadcastReceiver() {

    @Inject
    lateinit var manager: ReminderManager

    override fun onReceive(context: Context, intent: Intent) {
        (context.applicationContext as CoreyApp).appComponent.inject(this)

        if (manager.shouldScheduleWeighReminder()) {
            manager.postWeighNotification(context)
                .subscribe({
                    Timber.d("Weigh notification successfully posted!")
                }, { throwable ->
                    Timber.e(throwable)
                })
        }
        if (manager.isWorkoutReminderEnabled) {
            manager.postWorkoutNotification(context)
                .subscribe({
                    Timber.d("Workout notification successfully posted!")
                }, { throwable ->
                    Timber.e(throwable)
                })
        }
    }
}
